kaniko/kaniko.yaml

```diff
@@ -35,7 +36,17 @@ spec:
     - /kaniko/executor
     - $(params.EXTRA_ARGS)
     - --dockerfile=$(params.DOCKERFILE)
-    - --context=/workspace/source/$(params.CONTEXT)  # The user does not need to care the workspace and the source.
-    - --destination=$(resources.outputs.image.url)
+    - --context=$(workspaces.source.path)/$(params.CONTEXT)  # The user does not need to care the workspace and the source.
+    - --destination=$(params.IMAGE)
+    - --oci-layout=$(workspaces.source.path)/image-digest
     securityContext:
       runAsUser: 0
+  - name: write-digest
+    workingDir: $(workspaces.sources.path)
+    image: imagedigestexport-and-jq # FIXME(vdemeester) change with updated image
+    script: |
+      # output of imagedigestexport [{"name":"image","digest":"sha256:eed29..660"}]
+      /ko-app/imagedigestexporter \
+        -images [{"name":"$(params.IMAGE)","type":"image","url":"$(params.IMAGE)","digest":"","OutputImageDir":"$(workspaces.source.path)/image-digest"}] \
+        -terminationMessagePath=/tmp/digest
+      cat  | jq '.[0].digest' | tee /tekton/results/IMAGE_DIGEST
```

Comment:
Does `jq` add any newlines after the digest string when it's piped like this?  Suggest adding a `tr -d '\n'` if so to make sure the result the controller receives is just the digest value minus whitespace.

Reply:
also, should this be `cat /tmp/digest`?

Reply:
> Does `jq` add any newlines after the digest string when it's piped like this? Suggest adding a `tr -d '\n'` if so to make sure the result the controller receives is just the digest value minus whitespace.

`jq -c ` should do the trick too